### DISCUSSION TITLE
Not sure how to stream large amounts of data when the total size is unknown.

### DISCUSSION THREAD

**Discussion by thomasjwebb** [uNetworking/uWebSockets#1562] [Q&A]
I can see in examples and documentation how to stream large amounts of data when the total amount is known - use `tryEnd` and `onWritable`. But tryEnd will close the stream when the total amount is reached and I need to start sending data down before I know what the total size will be. It would be useful if there was a `tryWrite` that worked similarly but didn't require knowing the total size right away.

**Comment by uNetworkingAB** [comment]
tryWrite is a good opinion, yes we should have it, but so far we only have write. You can stream small chunks with write of, let's  say 4kb a time, checking the return of write, stopping when full, continuing in onWritable.

**Comment by uNetworkingAB** [comment]
So write will use Transfer-Encoding: chunked and you can write as much you want until calling end

**Comment by HagaiVinik** [comment]
@uNetworkingAB Sorry for pushing myself into your conversation, So if I'm understanding correctly, when `write()` fails to write to socket, I should `pause()` it, and then when `onWriteable()` is getting triggered I should try to write and then `resume()` it?

**Reply by uNetworkingAB** [reply]
pause and resume is not used here. you can read the docs and see examples
